phonebook-ui/src/app/common/service/user.service.ts

```diff
@@ -1,24 +1,30 @@
-import {Injectable} from '@angular/core';
-import {HttpClient} from "@angular/common/http";
-import {User} from "../model/user";
+import { Injectable } from '@angular/core'
+import { HttpClient } from '@angular/common/http'
+import { Observable } from 'rxjs'
+import { User } from '../interface/user.interface'
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserService {
 
-  private host = 'http://localhost:8080/';
-  private registerUrl = `${this.host}api/v1/user/registration`;
-  private registerConfirmUrl = `${this.host}api/v1/user/confirmation?token=`;
-
   constructor(private http: HttpClient) {
   }
 
   registerNewUser(user: User) {
-    return this.http.post<User>(this.registerUrl, user);
+    return this.http.post<User>('/api/user/registration', user)
   }
 
   sendRequestToRegisterConfirm(token: string) {
-    return this.http.get(`${this.registerConfirmUrl}${token}`);
+    return this.http.get(`/api/user/confirmation?token=${token}`)
+  }
+
+  recoveryPassword(user: User): Observable<User> {
+    return this.http.post<User>('/api/reset-password', user)
   }
+
+  resetPassword(user: User, token: string): Observable<User> {
```

Comment:
great . And please no string expressions in the code - make constants either in the class or in the file below the class.

Reply:
where are string expressions?